sdk/eventhub/Microsoft.Azure.EventHubs.Processor/src/PartitionContext.cs

```diff
@@ -123,6 +123,12 @@ public Task CheckpointAsync()
             Checkpoint capturedCheckpoint;
             lock(this.ThisLock)
             {
+                // Don't checkpoint for start of stream so that we can still respect initial offset provider.
+                if (this.Offset == "-1")
```

Comment:
Should we consider adding a test or two for this?

Reply:
I will add in the next PR :) Thx Jesse!